Screen_OLED_Test/OLEDScreen.cpp

```diff
@@ -0,0 +1,287 @@
+        oldHighlight = stateHighlight;
+        break;
+      case 3:
+        u8g2.drawXBMP( 43,  35,  21, 29, highlight_bits);
+        oldHighlight = 0;
+        break;
+      default:
+        u8g2.drawXBMP( 0,  35,  21, 29, highlight_bits);
+        oldHighlight = stateHighlight + 1;
+        break;
+    }
+}
+
+void batteryStatus() {
+  if (digitalRead(button2) == LOW) { // Temporary button switching battery
+    Serial.print("switch\n");
+    delay(100);
+    stateBattery = oldBattery + 1;
+    delay(100);
+  }
+
+  switch (stateBattery) { // switching battery percentages
+    case 1:
+      u8g2.drawXBMP( 17,  9,  6, 12, status_battery_bits);
+      oldBattery = stateBattery;
+      break;
+    case 2:
+      u8g2.drawXBMP( 17,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 23,  9,  6, 12, status_battery_bits);
+      oldBattery = stateBattery;
+      break;
+    case 3:
+      u8g2.drawXBMP( 17,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 23,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 29,  9,  6, 12, status_battery_bits);
+      oldBattery = stateBattery;
+      break;
+    case 4:
+      u8g2.drawXBMP( 17,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 23,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 29,  9,  6, 12, status_battery_bits);
+      u8g2.drawXBMP( 35,  9,  6, 12, status_battery_bits);
+      oldBattery = 0;
+      break;
+    default:
+      u8g2.drawXBMP( 17,  9,  6, 12, status_battery_bits);
+      oldBattery = stateBattery + 1;
+      break;
+  }
+}
+
+void arrowUp() {
+  // Arrow Up Higlight
+  if (digitalRead(buttonUp) == LOW) {
+    u8g2.drawXBMP( 74, 47, 12, 13, arrow_up_highlight_bits);
+  }
+}
+
+void arrowDown() {
+  // Arrow Down Highlight
+  if (digitalRead(buttonDown) == LOW) {
+    u8g2.drawXBMP( 108, 46, 12, 13, arrow_down_highlight_bits);
+  }
+}
+
+void elements() {
+    speed = map(analogRead(A0), 0, 1023, 0, 100); // Converting analog readings from 0-1023 to 0-100
+    itoa(speed, speed_string, 10); //integer converted into char with the 10 decimal value
+    speed_string_lenght = strlen(speed_string); // Computes the length of the string str up to
+
+    int savedHours = readInt(address);
+    int savedMinutes = readInt(address + 2);
+    int savedSeconds = readInt(address + 4);
+
+    u8g2.firstPage();
+    do {
+        for (int i = 0; i < speed_string_lenght; i++) {
+            u8g2.drawXBMP((97 - speed_string_lenght * 8) + 18*i, 15, 16, 28, bitmap_allArray[speed_string[i] - 48]);
+            /* drawing speed_string, subtracting the decimal value of 48 (char value of 0)
+      x coordinate is calculated so that it stays centered (center value - number of digits * (half the pixels of individual digits) ) + width of the character (+2 for spacing)*i
+      */
+        }
+        modeHighlight();
+        batteryStatus();
+        arrowUp();
+        arrowDown();
+        // Drawing Time
+        if (currentMode != MODE_SHOW_TIME) {
```

Comment:
Maybe explain a little bit what you are doing here exactly. Of course you are drawing triangles. But why?